pkg/kube/client.go

```diff
@@ -369,13 +371,64 @@ func batchPerform(infos ResourceList, fn func(*resource.Info) error, errs chan<-
 			kind = currentKind
 		}
 		wg.Add(1)
-		go func(i *resource.Info) {
-			errs <- fn(i)
+
+		backoff := wait.Backoff{
+			Duration: time.Duration(10) * time.Millisecond,
+			Factor:   5,
+			Jitter:   5,
+			Steps:    3,
+		}
+		go func(info *resource.Info) {
+			var fnErr error
+
+			// Retry according to backoff if this is a ResourceQuota conflict
+			_ = retry.RetryOnConflict(
```

Comment:
The error should be returned here

Reply:
That's the issue which I see with https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict, though:
~~~
RetryOnConflict is used to make an update to a resource when you have to worry about conflicts caused by other code making unrelated updates to the resource at the same time. fn should fetch the resource to be modified, make appropriate changes to it, try to update it, and return (unmodified) the error from the update function. On a successful update, RetryOnConflict will return nil. If the update function returns a "Conflict" error, RetryOnConflict will wait some amount of time as described by backoff, and then try again. On a non-"Conflict" error, or if it retries too many times and gives up, RetryOnConflict will return an error to the caller. 
~~~

If the code is to catch all Conflicts, then it would also have to run a Get to refresh the resource in question (is this even possible at this point here in the code?). Without a refresh / get of the resource, any other conflict would just loop around and would repeat right away. 
With the resourcequota issue, though, it is sufficient to simply retry as the resource itself is not in Conflict, but the resourcequota.

That's why I have the anonymous function return nil in the case of other errors because I do not want RetryOnConflict to take action if it's not specifically a conflict on resourcequota.

Regardless of that, the anonymous function sets `err` in line:
~~~
err = fn(info)
~~~
 
And that's then appended to the channel: 
~~~
errs <- err
~~~

I'm open to other suggestions, I can't wrap my head around that one otherwise.

Reply:
Alternatively, if we can guarantee that the functions which are passed to fn() do a refresh, this could be greatly simplified to:
~~~
                go func(info *resource.Info) {
                        // Retry according to backoff if this is a ResourceQuota conflict
                        errs <- retry.RetryOnConflict(
                                retry.DefaultRetry,
                                func() error {
                                        return fn(info)
                                })
                        wg.Done()
                }(info)
~~~

And the helper function would not be needed, either. Do I see this right?